src/connection.c

```diff
@@ -655,3 +655,40 @@ void close_and_unref_connection(noPollConn *conn)
     }
 }
 
+void write_conn_in_prog_file (const char *msg, time_t timestamp)
```

Comment:
As discussed, please use start, stop key and current timestamp value instead of +15mins